packages/react/src/utils/usePanelResize.ts

```diff
@@ -0,0 +1,36 @@
+'use client';
+import * as React from 'react';
+import { ownerWindow } from './owner';
+
+export function usePanelResize(
+  panelRef: React.RefObject<HTMLElement | null>,
+  setDimensions: (dimensions: { height: number; width: number }) => void,
+  shouldRender: boolean,
+) {
+  React.useEffect(() => {
+    const panel = panelRef.current;
+    if (!panel) {
+      return undefined;
+    }
+
+    // Ensures the panel will expand to the correct height when the window is resized.
+    // This prevents content from being cut off or the panel not fitting to the content.
+    function handleWindowResize() {
+      if (panel) {
+        const originalHeight = panel.style.height;
+        const originalWidth = panel.style.width;
+        panel.style.height = 'auto';
+        panel.style.width = 'auto';
+        setDimensions({ height: panel.scrollHeight, width: panel.scrollWidth });
```

Comment:
The idea of removing the CSS variable when the transition completes (so it uses implicit `auto`) may be a better idea, depending if we want the CSS variable present with the values all the time or not